devApps/AdalAndroidTestApp/MainActivity.cs

```diff
@@ -32,6 +32,7 @@
 using Android.Widget;
 using Android.OS;
 using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using Runnable = Java.Lang.Runnable;
```

Comment:
Seems like most of the code here should be synched to the ui thread. This behavior could be moved to the textview control. 

Also: What about other platforms? Do they do something similar?